### ISSUE TITLE
Add netstandard2.0 support

### ISSUE THREAD

**Issue by ahoka** [lecaillon/Evolve#80]
I see that currently the nuget is only supporting netcoreapp2.0, but not netstandard2.0, which would be much better. Is there any explicit reason for this?

**Comment by lecaillon** [comment]
To connect to the database, Evolve loads dynamically at build-time the database driver asssemblies of your application (Npgsql, Mysql.Data ... )
For .NET projects it uses `AppDomain` that is not supported by .NET Core projects (even if its in the netstandard2.0... https://github.com/dotnet/standard/issues/456).
For .NET Core projects it uses `CustomAssemblyLoader`, that is part of the netstandard1.5. Honestly I never tested it to load .NET apps, but I think it will not work either.
In short, the reason is linked to the tricky part of dynamically asssembly loading

**Comment by lecaillon** [comment]
You should be able to do it with the existing package. Try it and give me some feedback.
Make a package for this specific need is not easy to explain to others users. A package without all the features does not seem a good idea, without thinking about the additional work...

**Comment by mfjerome** [comment]
Would also like Evolve to be Net Standard eventually. Here is a new guidance published my Microsoft: https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting .

I am not sure how dynamic assembly loading is handled in Evolve, but if it might be of any help, we currently use [Steeltoe Connector](https://github.com/SteeltoeOSS/Connectors) which is a net standard component that loads the correct DB driver required at runtime by the application to connect to a Cloud Foundry database service.

**Comment by lecaillon** [comment]
@mfjerome Thx for the Steeltoe link. I'm going to look at it.

**Comment by lecaillon** [comment]
I checked the Steeltoe repo and I think they only provide a way to configure the database driver in a fancy way. It is what they call a connector. In the other hand, Evolve has to instanciate the right driver wihtout any reference to them. At build time it loads dynamically the databse driver assemblies of the application being built (.NET or NET Core) and it is the hard part not so netstandard-ish :)

Anyway I'm currently working on Evolve 2.0 to get rid of that part (dynamic loading assemblies) that complicates a lot the development of new features. I'm planning to divide Evolve in 3 parts:
- `Evolve`: a **netstandard2.0** library you have to add to your project in order to migrate your SQL scripts.
- `Evolve.Cli`: a standalone CLI that allows you to run Evolve outside your application. This CLI will have a dependency on Evolve and will embed the 5 database drivers supported. It will support Windows, Linux and MacOS. Its size +20Mo (that is the downside obviously).
- `Evolve.MSBuild`: a package that allows you to run Evolve at build-time. It will add the `Evolve.targets` and the MSBuild task needed and will lay on the Evolve.CLI.

**Comment by lecaillon** [comment]
Addressed in #99
